### PR TITLE
Return application/json as content-type header

### PR DESCRIPTION
**Pull request by pradithya**
**What this PR does / why we need it**:
Set `application/json` as content-type header in response. 

**Which issue(s) this PR fixes**:
Fixes issue where the response type header is incorrectly set to `text/html`

```
* upload completely sent off: 27 out of 27 bytes
< HTTP/1.1 200 OK
< content-length: 20
< content-type: text/html; charset=UTF-8
< date: Thu, 19 Dec 2019 12:54:43 GMT
< server: istio-envoy
< x-envoy-upstream-service-time: 6
```

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/615)

### CODE REVIEW

File: python/kfserving/kfserving/handlers/http.py [hunk 1]
```diff
@@ -49,14 +49,21 @@ def post(self, name: str):
...
         model = self.get_model(name)
-        request = self.validate(self.request)
+        try:
+            body = json.loads(self.request.body)
+        except json.decoder.JSONDecodeError as e:
+            raise tornado.web.HTTPError(
+                status_code=HTTPStatus.BAD_REQUEST,
+                reason="Unrecognized request format: %s" % e
+            )
+        request = self.validate(body)
```
Comment (yuzisun) [inline python/kfserving/kfserving/handlers/http.py]: should `validate` also be called after `preprocess` like in predict ? The request may come in with arbitrary data plane but after preprocess it should conform to KFServing data plane for prediction.

Comment (pradithya) [reply python/kfserving/kfserving/handlers/http.py]: Fixed it.

File: python/kfserving/test/test_server.py [hunk 2]
```diff
@@ -48,7 +51,17 @@ def app(self):  # pylint: disable=no-self-use
...
+        assert resp.body == b'{"predictions": [[1, 2]]}'
+        assert resp.headers['content-type'] == "application/json; charset=UTF-8"
+
+    async def test_explain(self, http_server_client):
+        resp = await http_server_client.fetch('/v1/models/TestModel:explain',
+                                              method="POST",
                                               body=b'{"instances":[[1,2]]}')
         assert resp.code == 200
         assert resp.body == b'{"predictions": [[1, 2]]}'
+        assert resp.headers['content-type'] == "application/json; charset=UTF-8"
```
Comment (yuzisun) [inline python/kfserving/test/test_server.py]: lint is complaining the newline

Comment (pradithya) [reply python/kfserving/test/test_server.py]: Fixed it

**Comment by pradithya** [comment]
@yuzisun I believe it's ready to be merged, any more suggestion?

**Comment by yuzisun** [comment]
Awesome!

/lgtm
/approve

**Pull request merged**